src/Discord.Net.WebSocket/Entities/Invites/SocketInvite.cs

```diff
@@ -0,0 +1,137 @@
+using Discord.Rest;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Model = Discord.API.Gateway.InviteCreateEvent;
+
+namespace Discord.WebSocket
+{
+    [DebuggerDisplay(@"{DebuggerDisplay,nq}")]
+    public class SocketInvite : SocketEntity<string>, IInviteMetadata
+    {
+         private long _createdAtTicks;
+
+        /// <inheritdoc />
+        public ulong ChannelId { get; private set; }
+        /// <summary>
+        ///     Gets the channel where this invite was created.
+        /// </summary>
+        public SocketGuildChannel Channel { get; private set; }
+        /// <inheritdoc />
+        public ulong? GuildId { get; private set; }
+        /// <summary>
+        ///     Gets the guild where this invite was created.
+        /// </summary>
+        public SocketGuild Guild { get; private set; }
+        /// <inheritdoc />
+        ChannelType IInvite.ChannelType => throw new NotImplementedException();
+        /// <inheritdoc />
+        string IInvite.ChannelName => throw new NotImplementedException();
+        /// <inheritdoc />
+        string IInvite.GuildName => throw new NotImplementedException();
```

Comment:
These props *should* be able to be fulfilled using the cache, and aren't potentially blocked behind something like privileges or needing something from REST, so, should always be available to grab from client state. Can easily grab these from the discord client in the ctor.